### ISSUE TITLE
Github template source failing with `some refs were not updated`

### ISSUE THREAD

**Issue by rory-lamendola** [argoproj/argo-events#1120]
**Describe the bug**
The github template source starts failing with `failed to fetch. err: some refs were not updated` after about 30 minutes to an hour after the sensor is created.  This happens even when not pulling from a particular ref or branch, just pulling from master.  From what I understand, setting the ref should result in a shallow clone and fetch, so not sure why it wouldn't be able to update that ref.  And it makes even less sense for that to fail when removing that ref, and just using the default master branch. 

**To Reproduce**

Sensor.yaml:
```
---
apiVersion: argoproj.io/v1alpha1
kind: Sensor
metadata:
  name: github
  namespace: argo
spec:
  template:
    serviceAccountName: argo-events-sa
    containers:
      - volumeMounts:
          - mountPath: /git/shared
            name: shared
    volumes:
      - name: shared
        emptyDir: {}
  dependencies:
    - name: webhook-dep
      eventSourceName: github
      eventName: webhook
  triggers:
    - template:
        name: workflow-trigger
        k8s:
          group: argoproj.io
          version: v1alpha1
          resource: workflows
          operation: create
          source:
            git:
              url: "git@github.com:petal-engineering/shared.git"
              cloneDirectory: "/git/shared"
              sshKeySecret:
                name: github
                key: ssh-private-key
              filePath: "ci/argo/ci.yml"
      parameters:
        - src:
            dependencyName: webhook-dep
            dataKey: body.pull_request.head.ref
          dest: k8s.source.git.branch
```

I'm not able to reproduce it using the public argo workflow repo instead of our private repo.

**Expected behavior**
I expect the sensor to trigger the yaml that exists on the given branch on github.  

**Screenshots**

**Environment (please complete the following information):**
 - Kubernetes: 1.18.9 (eks)
 - Argo: v0.16.7
 - Argo Events: v1.2.3

**Additional context**

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by whynowy** [comment]
@rory-lamendola could you remove `cloneDirectory` from the spec (and the `volume` and `volumeMount` are not needed), and see if the issue still exits?

**Comment by rory-lamendola** [comment]
@whynowy So we think we found the issue.  Digging through the debug logs, I found that the error seemed to be coming from go-git, with the real error being that we need to force pull.  We figured out that our deploy process updates tags on branches of our git repo, which seems to explain why go-git needed to force pull.  I updated the source code on our k8 cluster to add in the force flag, and it seems to be working (at least for the last day).  Was going to open a pr with a configuration option to allow for force pulling, if that works with y'all
